client/lib/checkout/index.js

```diff
@@ -23,5 +23,5 @@ export function getExitCheckoutUrl( cart, siteSlug ) {
 		url = '/design/';
 	}
 
-	return url + siteSlug;
+	return siteSlug ? url + siteSlug : url;
```

Comment:
This seems a mostly reasonable solution from what I can see, if the passed in siteSlug is falsey, just don't append.